src/native/fs/fs_napi.cpp

```diff
@@ -1140,126 +1301,19 @@ struct LinkFileAt : public FSWrapWorker<FileWrap>
     }
 };
 
-struct FileGetxattr : public FSWrapWorker<FileWrap>
-{
-    std::map<std::string, std::string> _md;
-    FileGetxattr(const Napi::CallbackInfo& info)
-        : FSWrapWorker<FileWrap>(info)
-    {
-        Begin(XSTR() << "FileGetxattr " << DVAL(_wrap->_path));
-    }
-    virtual void Work()
-    {
-        int fd = _wrap->_fd;
-        // Serves as a "lock" mechanism when reading the xattrs of a file
-        // Since this is non atomic operation, we abort if ctime of file changed
-        struct stat stat_res;
-        auto r = fstat(fd, &stat_res);
-        if (r) {
-            SetSyscallError();
-            return;
-        }
-        auto start_ctime = stat_res.st_ctime;
-
-        #ifdef __APPLE__
-            ssize_t buf_len = flistxattr(fd, NULL, 0, 0);
-        #else
-            ssize_t buf_len = flistxattr(fd, NULL, 0);
-        #endif
-
-        // No xattr, nothing to do  
-        if (buf_len == 0) return;
-
-        if (buf_len == -1) {
-            SetSyscallError();
-            return;
-        }
-
-        Buf buf(buf_len);
-
-        #ifdef __APPLE__
-            buf_len = flistxattr(fd, buf.cdata(), buf_len, 0);
-        #else
-            buf_len = flistxattr(fd, buf.cdata(), buf_len);
-        #endif
-
-        // No xattr, nothing to do  
-        if (buf_len == 0) return;
-
-        if (buf_len == -1) {
-            SetSyscallError();
-            return;
-        }
-
-        while (buf.length()) {
-            std::string key(buf.cdata());
-
-            #ifdef __APPLE__
-                ssize_t value_len = fgetxattr(fd, key.c_str(), NULL, 0, 0, 0);
-            #else
-                ssize_t value_len = fgetxattr(fd, key.c_str(), NULL, 0);
-            #endif
-
-            if (value_len == -1) {
-                SetSyscallError();
-                return;
-            }
-
-            if (value_len > 0) {
-                Buf val(value_len);
-
-                #ifdef __APPLE__
-                    value_len = fgetxattr(fd, key.c_str(), val.cdata(), value_len, 0, 0);
-                #else
-                    value_len = fgetxattr(fd, key.c_str(), val.cdata(), value_len);
-                #endif
-
-                if (value_len == -1) {
-                    SetSyscallError();
-                    return;
-                }
-                
-                _md[key] = std::string(val.cdata(), value_len);
-            } else if (value_len == 0) {
-                _md[key] = "";
-            }
-
-            buf.slice(key.size() + 1, buf.length());
-        }
-
-        r = fstat(fd, &stat_res);
-        if (r) {
-            SetSyscallError();
-            return;
-        }
-        auto end_ctime = stat_res.st_ctime;
-        if (start_ctime != end_ctime) {
-            // Cancel on ctime change due to non atomic nature of this call
-            SetError(XSTR() << "FileGetxattr: " << DVAL(_wrap->_path) << " cancelled due to ctime change");
-        }
```

Comment:
I think this was accidentally removed - this is a check that the inode ctime was not changed in between the calls to get the xattrs, to make sure we don't race with another client making changes to the attributes.